isisd/isis_zebra.c

```diff
@@ -53,6 +53,8 @@
 
 struct zclient *zclient = NULL;
 
+DEFINE_HOOK(isis_if_new_hook, (struct interface *ifp), (ifp))
```

Comment:
It is necessary for isis_te part to set TE link parameter when interface goes up. In previous version `isis_mpls_te_update()` function was call by ` isis_zebra_link_params()` function when link parameters change. Unfortunately this will not cover all cases. By using this new hook, we guarantee that TE parameters are advertised only when interface goes up and when link param changes.

Reply:
Ok, marking as resolved as well.